coderd/workspaceapps/proxy.go

```diff
@@ -176,7 +180,10 @@ func (s *Server) handleAPIKeySmuggling(rw http.ResponseWriter, r *http.Request,
 	}
 
 	// Exchange the encoded API key for a real one.
-	token, err := s.AppSecurityKey.DecryptAPIKey(encryptedAPIKey)
+	var payload EncryptedAPIKeyPayload
+	err := jwtutils.Decrypt(ctx, s.APIKeyEncryptionKeycache, encryptedAPIKey, &payload, jwtutils.WithDecryptExpected(jwt.Expected{
+		Time: time.Now(),
+	}))
```

Comment:
If we get an error here, we redirect to a static error page, which is bad news if a user gets caught with an unsupported JWE without a key ID.  

Maybe we should detect a missing key ID and just strip the query parameter?

I guess you'd have to be pretty unlucky to be doing this API key dance _right_ as Coder is being upgraded. So, maybe it's not worth doing. WDYT?

Reply:
Hmm yeah, pretty unlikely and will only result in a one-off error right?

Reply:
I think so